packages/core/content-releases/admin/src/components/ReleaseActionOptions.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+
+import {
+  FieldInput,
+  FieldLabel,
+  VisuallyHidden,
+  Field,
+  type FieldProps,
+} from '@strapi/design-system';
+import styled from 'styled-components';
+
+interface FieldWrapperProps extends FieldProps {
+  actionType: 'publish' | 'unpublish';
+}
+
+const getBorderLeftRadiusValue = (actionType: FieldWrapperProps['actionType']) => {
+  return actionType === 'publish' ? 1 : 0;
+};
+
+const getBorderRightRadiusValue = (actionType: FieldWrapperProps['actionType']) => {
+  return actionType === 'publish' ? 0 : 1;
+};
```

Comment:
Maybe try putting a global radius on the ReleaseActionOptions parent with overflow hidden?

Reply:
Not sure I understand, wouldn't that mean putting styles outside the component?

I wouldn't say I am extremely proud of what I have here but also not sure what is wrong with it...

Reply:
It definitely works, it's just a lot of code and logic, so it took me a moment to understand what was going on. I'm trying to explore simpler solutions.

My idea is:

```tsx
export const ReleaseActionOptions = ({ selected, handleChange }: ActionOptionProps) => {
  return (
    <Box hasRadius style={{ overflow: 'hidden' }}> // 👈 changes here
      <ActionOption actionType="publish" selected={selected} handleChange={handleChange} />
      <ActionOption actionType="unpublish" selected={selected} handleChange={handleChange} />
    </>
  );
};
```

with this I believe you shouldn't need any border radius inside ActionOption

Reply:
ohhh sorry I misunderstood what you meant by parent 👍 . Ok I'll try this

Reply:
This does work nicely to handle the border radius, but where it gets tricky is maintaining the border color on the active selection (1/2 the input).

Maintaining as is gives a weird effect where part of the border radius is cut off:

![Screenshot 2023-11-29 at 11 47 09](https://github.com/strapi/strapi/assets/26598053/ffddef0b-191b-4253-ba31-22646f8806ed)

I think I'll leave it as is for now. We can always revisit it.